archetypes/Pools/index.tsx

```diff
@@ -43,54 +44,69 @@ export const Browse: React.FC = () => {
+                    return true;
+                default:
+                    return !!pool.name && pool.name.split('-')?.[0] === state.leverageFilter;
+            }
+        },
+        [state.leverageFilter],
+    );
 
-    const leverageFilter = (pool: BrowseTableRowData): boolean => {
-        switch (state.leverageFilter) {
-            case LeverageEnum.All:
-                return true;
-            default:
-                return !!pool.name && pool.name.split('-')?.[0] === state.leverageFilter;
-        }
-    };
-    const searchFilter = (pool: BrowseTableRowData): boolean => {
-        const searchString = state.search.toLowerCase();
-        return Boolean(
-            pool.name.toLowerCase().match(searchString) ||
-                pool.shortToken.symbol.toLowerCase().match(searchString) ||
-                pool.longToken.symbol.toLowerCase().match(searchString) ||
-                pool.market.toLowerCase().match(searchString),
-        );
-    };
+    const searchFilter = useCallback(
+        (pool: BrowseTableRowData): boolean => {
+            const searchString = state.search.toLowerCase();
+            return Boolean(
+                pool.name.toLowerCase().match(searchString) ||
+                    pool.shortToken.symbol.toLowerCase().match(searchString) ||
+                    pool.longToken.symbol.toLowerCase().match(searchString) ||
+                    pool.market.toLowerCase().match(searchString),
+            );
+        },
+        [state.search],
+    );
 
-    const sorter = (poolA: BrowseTableRowData, poolB: BrowseTableRowData): number => {
-        switch (state.sortBy) {
-            case SortByEnum.TotalValueLocked:
-                return poolB.tvl - poolA.tvl;
-            case SortByEnum.MyHoldings:
-                return poolB.myHoldings - poolA.myHoldings;
-            default:
-                return 0;
-        }
-    };
+    const sorter = useCallback(
+        (poolA: BrowseTableRowData, poolB: BrowseTableRowData): number => {
+            switch (state.sortBy) {
+                case SortByEnum.TotalValueLocked:
+                    return poolB.tvl - poolA.tvl;
+                case SortByEnum.MyHoldings:
+                    return poolB.myHoldings - poolA.myHoldings;
+                default:
+                    return 0;
+            }
+        },
+        [state.sortBy],
+    );
 
-    const filteredTokens = tokens
-        .filter((pool) => marketFilter(pool.name, state.marketFilter))
-        .filter(leverageFilter)
-        .filter(searchFilter);
-    const sortedFilteredTokens = filteredTokens.sort(sorter);
+    const filteredTokens = useMemo(
+        () =>
+            tokens
+                .filter((pool) => marketFilter(pool.name, state.marketFilter))
+                .filter(leverageFilter)
+                .filter(searchFilter)
+                .sort(sorter),
+        [state.marketFilter, state.leverageFilter, state.search, state.sortBy, tokens],
```

Comment:
should this have `pool.name`, `state.marketFilter`, `leverageFilter`, `searchFilter` and `sorter` as deps instead? looks like it will work but imagining dependencies as a graph this would be a child instead of a siblings to the filters/sorter

Reply:
pool is passed in as an arg, so it goes filter the pools, none of the filters have changed return the same thing

Reply:
is it guaranteed that `leveragedFilter`, `searchFilter` and `sorter` will have finished changing (as a result of `state.leverageFilter`, `state.search`, and `state.sortBy` changing) by the time this changes?

Reply:
yeah since they are callbacks defined above this one, theyll get updated first if any of the state changes